### ISSUE TITLE
FrontEnd: CLI get stuck for CREATE TABLE query

### ISSUE THREAD

**Issue by sumittal** [risingwavelabs/risingwave#6512]
### Describe the bug

Not able to create the table after the DB get few MVs.
image tag -  v0.1.14.00
Logs - https://ap-southeast-1.console.aws.amazon.com/cloudwatch/home?region=ap-southeast-1#logsV2:log-groups/log-group/$252Faws$252Fecs$252Fworkflow-engine-lg/log-events/rls-apse1-mgmt$252Fworkflow-engine-task$252F0e26caadf93d487896e57e4712384481

### To Reproduce

We have the below tenant:
`psql "postgres://dev:dev@rls-apse1-eks-a.risingwave-cloud.xyz:4566/dev?options=--tenant%3Drwc-6-release-testing"`

And stressed the DB with some load and having few MVs:

```
dev=> SHOW MATERIALIZED VIEWS;
     Name
--------------
 nexmark_q7
 nexmaark_q71
 nexmaark_q7
(3 rows)
```

Then tried to create few tables as below:

> dev=> create table t1 (v1 int, v2 int);
> 
> ERROR:  QueryError: internal error: table or source is in creating procedure
> dev=> insert into t1 values(1, 2);
> ERROR:  QueryError: Catalog error: table not found: t1

It stuck for forever.

**Comment by sushantgupta** [comment]
hit the same issue on today's build.
```
CREATE MATERIALIZED VIEW nexmark_q5
AS
SELECT
    AuctionBids.auction, AuctionBids.num
FROM (
    SELECT
        bid.auction,
                count(*) AS num,
        window_start AS starttime
    FROM
        HOP(bid, date_time, INTERVAL '2' SECOND, INTERVAL '10' SECOND)
    GROUP BY
        window_start,
                bid.auction
) AS AuctionBids
JOIN (
        SELECT
                max(CountBids.num) AS maxn,
                CountBids.starttime_c
        FROM (
                SELECT
                        count(*) AS num,
                        window_start AS starttime_c
                FROM HOP(bid, date_time, INTERVAL '2' SECOND, INTERVAL '10' SECOND)
                GROUP BY
                        bid.auction,
                        window_start
                ) AS CountBids
        GROUP BY
                CountBids.starttime_c
        ) AS MaxBids
ON
        AuctionBids.starttime = MaxBids.starttime_c AND
        AuctionBids.num >= MaxBids.maxn;



^C^C^C^C^C^C^C^C


^CCancel request sent
Cancel request sent
^C^CCancel request sent
^CCancel request sent
^CCancel request sent
Cancel request sent

```

**Comment by sumittal** [comment]
So, the pull request https://github.com/risingwavelabs/risingwave/pull/6634 has been cherry picked and merged to the release branch i.e. v0.1.14-rc.

**Comment by yezizp2012** [comment]
There're two possible ways might cause this problem:
1. deadlock in LRU cache, should be fixed in #6634 .
2. the system is in recovering with some of the CNs lost forever like #6639 . After #6651, it will report error rather than get stuck. Could you please check all components status especially compute nodes and provide more logs.

Could you please share the setup details and logs informations ? @HuaHuaY , PTAL.

**Comment by lmatz** [comment]
> 2\. feat(meta): reject ddl with streaming jobs when recovering #6651

Does it mean that https://github.com/risingwavelabs/risingwave/pull/6651 should also be merged into v0.1.14-rc?

**Comment by huangjw806** [comment]
#6634 doesn't fully cover the issue？cc @BugenZhao

**Comment by yezizp2012** [comment]
> > 2. feat(meta): reject ddl with streaming jobs when recovering #6651
> 
> Does it mean that #6651 should also be merged into v0.1.14-rc?

Yes, I think so. It can help to expose the problem if indeed it's cause by some lose of CNs.

**Comment by sushantgupta** [comment]
@yezizp2012 
I couldn't find any related logs, so I am attaching the all the logs:

```
╰─> kubectl logs -n rwc-3-sushant3node risingwave-frontend-6d7bfb75c9-jq5fj
launching frontend node
2022-11-30T07:57:41.776482Z  INFO risingwave_rt: setting tokio worker threads to 8
2022-11-30T07:57:41.777141Z  INFO risingwave_frontend::session: Starting frontend node with
frontend config FrontendConfig { server: ServerConfig { heartbeat_interval_ms: 1000, max_heartbeat_interval_secs: 600, connection_pool_size: 16 } }
batch config BatchConfig { worker_threads_num: None, developer: DeveloperConfig { batch_output_channel_size: 64, batch_chunk_size: 1024, stream_enable_executor_row_count: false, stream_enable_managed_cache: true, stream_connector_message_buffer_size: 16, unsafe_stream_hash_agg_cache_size: 65536, unsafe_stream_join_cache_size: 65536, unsafe_stream_extreme_cache_size: 1024, stream_chunk_size: 1024 } }
2022-11-30T07:57:41.777215Z  WARN risingwave_frontend::session: Client address is not specified, defaulting to host address
2022-11-30T07:57:41.777233Z  INFO risingwave_frontend::session: Client address is 10.0.5.84:4567
2022-11-30T08:27:30.679980Z  INFO risingwave_frontend::scheduler::hummock_snapshot_manager: Unpin epoch 3444680209530880 with RPC
2022-11-30T08:27:41.780845Z  INFO risingwave_frontend::scheduler::hummock_snapshot_manager: Unpin epoch 3444680733884416 with RPC
2022-11-30T08:28:11.781234Z  INFO risingwave_frontend::scheduler::hummock_snapshot_manager: Unpin epoch 3444681094266880 with RPC
2022-11-30T08:28:21.781513Z  INFO risingwave_frontend::scheduler::hummock_snapshot_manager: Unpin epoch 3444681274556416 with RPC
2022-11-30T08:28:51.781454Z  INFO risingwave_frontend::scheduler::hummock_snapshot_manager: Unpin epoch 3444681454714880 with RPC
2022-11-30T08:29:21.781470Z  INFO risingwave_frontend::scheduler::hummock_snapshot_manager: Unpin epoch 3444681754411008 with RPC
2022-11-30T08:29:41.781421Z  INFO risingwave_frontend::scheduler::hummock_snapshot_manager: Unpin epoch 3444682082025472 with RPC
2022-11-30T09:38:15.042409Z ERROR risingwave_frontend::session: failed to handle sql:
CREATE MATERIALIZED SOURCE person (
    "id" BIGINT,
    "name" VARCHAR,
    "email_address" VARCHAR,
    "credit_card" VARCHAR,
    "city" VARCHAR,
    "state" VARCHAR,
    "date_time" TIMESTAMP,
    "extra" VARCHAR,
    PRIMARY KEY (id)
) with (
    connector = 'nexmark',
    nexmark.table.type = 'Person',
    nexmark.split.num = '12',
    nexmark.min.event.gap.in.ns = '0'
) ROW FORMAT JSON;:
Catalog error: source with name person exists

╭─sushantgupta@Sushants-MacBook-Pro in ~ via 🐳 desktop-linux via 🐍 v3.10.8 on ☁️  (ap-southeast-1)
╰─>

╰─> kubectl logs -n rwc-3-sushant3node risingwave-meta-74c65786bb-5g87t
launching meta node
2022-11-30T07:57:00.700195Z  INFO risingwave_rt: setting tokio worker threads to 8
2022-11-30T07:57:00.701019Z  INFO risingwave_meta: Starting meta node with config MetaNodeConfig { streaming: StreamingConfig { barrier_interval_ms: 250, in_flight_barrier_nums: 40, checkpoint_frequency: 10, minimal_scheduling: false, worker_node_parallelism: 8, actor_runtime_worker_threads_num: None, total_memory_available_bytes: 66716327936, developer: DeveloperConfig { batch_output_channel_size: 64, batch_chunk_size: 1024, stream_enable_executor_row_count: false, stream_enable_managed_cache: true, stream_connector_message_buffer_size: 16, unsafe_stream_hash_agg_cache_size: 65536, unsafe_stream_join_cache_size: 65536, unsafe_stream_extreme_cache_size: 1024, stream_chunk_size: 1024 } } }
2022-11-30T07:57:00.701042Z  INFO risingwave_meta: Starting meta node with options periodic_compaction_interval_sec: 60, enable_compaction_deterministic: false
2022-11-30T07:57:00.701046Z  INFO risingwave_meta: Meta server listening at 0.0.0.0:5690
2022-11-30T07:57:00.704250Z  INFO risingwave_meta::barrier::schedule: Starting barrier scheduler with: checkpoint_frequency=10
2022-11-30T07:57:00.704272Z  INFO risingwave_meta::barrier: Starting barrier manager with: interval=250ms, enable_recovery=true, in_flight_barrier_nums=40
2022-11-30T07:57:00.704344Z  INFO risingwave_common_service::metrics_manager: Prometheus listener for Prometheus is set up on http://0.0.0.0:1250
2022-11-30T07:57:00.704514Z  WARN risingwave_common::util::epoch: New generate epoch is too close to the previous one.
2022-11-30T07:57:00.704610Z  WARN risingwave_common::util::epoch: New generate epoch is too close to the previous one.
2022-11-30T07:57:00.704698Z  INFO risingwave_meta::hummock::compaction_scheduler: Start compaction scheduler.
2022-11-30T07:57:12.301496Z  INFO risingwave_meta::hummock::compactor_manager: Added compactor session 1
2022-11-30T07:57:41.821164Z  INFO risingwave_meta::hummock::compactor_manager: Added compactor session 1
2022-11-30T08:27:28.657592Z  INFO risingwave_meta::stream::source_manager: spawning new watcher for source 1001
2022-11-30T08:27:28.765728Z  INFO risingwave_meta::stream::source_manager: spawning new watcher for source 1004
2022-11-30T08:27:28.997952Z  INFO risingwave_meta::stream::source_manager: spawning new watcher for source 1007
2022-11-30T08:28:01.535025Z  INFO risingwave_meta::hummock::vacuum: Finish vacuuming SSTs [1, 2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16, 17, 18, 20, 21, 22, 23, 24, 25, 26, 27, 28, 30, 31, 32, 33, 35, 36, 37, 38, 39, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 57, 58, 60, 61]
2022-11-30T08:28:30.922438Z  INFO risingwave_meta::hummock::vacuum: Finish vacuuming SSTs [9, 19, 29, 71, 73, 81, 83, 91, 93]
2022-11-30T08:29:01.092491Z  INFO risingwave_meta::hummock::vacuum: Finish vacuuming SSTs [59, 70, 104, 106, 108, 110, 111, 112, 113, 114, 115, 116, 119, 120, 122, 123, 124, 125, 129, 131, 135, 137, 139]
2022-11-30T08:29:31.019248Z  INFO risingwave_meta::hummock::vacuum: Finish vacuuming SSTs [55, 56, 72, 74, 77, 82, 84, 86, 92, 95, 103, 105, 107, 109, 133, 134, 136, 138, 140, 141, 142, 560]
2022-11-30T08:30:01.758113Z  INFO risingwave_meta::hummock::vacuum: Finish vacuuming SSTs [40, 62, 63, 64, 65, 67, 68, 69, 75, 76, 79, 85, 87, 94, 118, 121, 126, 127, 130, 132, 143, 144, 145, 146, 148, 150, 341, 344, 346, 348, 350, 362, 364, 370, 490, 571, 731, 732, 733, 734, 735, 736, 737, 739, 740, 1802, 1926, 1930, 2001, 2002, 2005, 2007, 2009, 2010, 2132, 2135, 2138]

╭─sushantgupta@Sushants-MacBook-Pro in ~ via 🐳 desktop-linux via 🐍 v3.10.8 on ☁️  (ap-southeast-1)
╰─> kubectl logs -n rwc-3-sushant3node risingwave-compactor-59c8544d76-6zxgd
2022-11-30T08:29:47.804275Z  INFO risingwave_storage::hummock::compactor: Finished compaction task in 5445.383908ms:
Compaction task id: 161, target level: 6
Compaction watermark: 3444682082025472
Compaction target_file_size: 33554432
Compaction # splits: 1
Compaction task status: Success
Compaction Sstables structure:
Level 5 ["[id: 740, 32385KB]", "[id: 1926, 32385KB]", "[id: 2001, 32385KB]", "[id: 2135, 32385KB]", "[id: 2138, 32385KB]"]
Level 6 ["[id: 144, 32385KB]", "[id: 146, 32385KB]", "[id: 341, 32385KB]", "[id: 344, 32385KB]", "[id: 346, 32385KB]", "[id: 348, 32385KB]", "[id: 350, 32385KB]", "[id: 362, 32385KB]", "[id: 364, 3896KB]", "[id: 145, 32385KB]"]
Compaction task output:
SstableInfo: id=2188, KeyRange=["000003ea000080000000000006c5000c3ceb1c260000","000003ea070080000000000ad205000c3ceb2ae70000"], size=32385KB
SstableInfo: id=2253, KeyRange=["000003ea070080000000000ad21c000c3ceb26ff0000","000003ea0e00800000000015a1be000c3ceb35a60000"], size=32385KB
SstableInfo: id=2364, KeyRange=["000003ea0e00800000000015a40e000c3ceb3b820000","000003ea15008000000000207549000c3ceb4e0f0000"], size=32385KB
SstableInfo: id=2369, KeyRange=["000003ea15008000000000207550000c3ceb4e0f0000","000003ea1c0080000000002b50eb000c3ceb54e50000"], size=32385KB
SstableInfo: id=2415, KeyRange=["000003ea1c0080000000002b50f2000c3ceb55df0000","000003ea2300800000000036c7ca000c3ceb686e0000"], size=32385KB
SstableInfo: id=2418, KeyRange=["000003ea2300800000000036c7d3000c3ceb78560000","000003ea2a00800000000047551f000c3ceb81200000"], size=32385KB
SstableInfo: id=2792, KeyRange=["000003ea2a008000000000475b80000c3ceb85690000","000003ea3200800000000007b7c4000c3ceb24110000"], size=32385KB
SstableInfo: id=2794, KeyRange=["000003ea3200800000000007b7dd000c3ceb26060000","000003ea390080000000001273bd000c3ceb30c30000"], size=32385KB
SstableInfo: id=2796, KeyRange=["000003ea3900800000000012760d000c3ceb34ab0000","000003ea400080000000001d8583000c3ceb463f0000"], size=32385KB
SstableInfo: id=2798, KeyRange=["000003ea400080000000001d859a000c3ceb492d0000","000003ea4700800000000028bf4d000c3ceb59c70000"], size=32385KB
SstableInfo: id=2799, KeyRange=["000003ea4700800000000028bf54000c3ceb50fd0000","000003ea4e00800000000033287c000c3ceb62920000"], size=32385KB
SstableInfo: id=2800, KeyRange=["000003ea4e008000000000332dcc000c3ceb61980000","000003ea55008000000000426535000c3ceb7c3f0000"], size=32385KB
SstableInfo: id=2911, KeyRange=["000003ea55008000000000426b81000c3ceb8bdd0000","000003ea5d00800000000004b578000c3ceb20290000"], size=32385KB
SstableInfo: id=2912, KeyRange=["000003ea5d00800000000004bbcc000c3ceb20290000","000003ea640080000000000f14e5000c3ceb31be0000"], size=32385KB
SstableInfo: id=2913, KeyRange=["000003ea640080000000000f14fc000c3ceb2cdb0000","000003ea6500800000000005f908000c3ceb23170000"], size=3853KB

2022-11-30T08:30:00.706358Z  INFO risingwave_storage::hummock::vacuum: Try to vacuum SSTs [40, 62, 63, 64, 65, 67, 68, 69, 75, 76, 79, 85, 87, 94, 118, 121, 126, 127, 130, 132, 143, 144, 145, 146, 148, 150, 341, 344, 346, 348, 350, 362, 364, 370, 490, 571, 731, 732, 733, 734, 735, 736, 737, 739, 740, 1802, 1926, 1930, 2001, 2002, 2005, 2007, 2009, 2010, 2132, 2135, 2138]
2022-11-30T08:30:01.758785Z  INFO risingwave_storage::hummock::vacuum: Finished vacuuming SSTs

```

**Comment by BugenZhao** [comment]
We haven't supported canceling creating materialized views yet, so pressing ^C does not help. :( Could you please confirm that we're stuck by starting a new psql session and typing "flush;"? If there's indeed no progress, could you also please provide the output of `risectl trace`?

**Comment by yezizp2012** [comment]
I can connect with `psql "postgres://dev:dev@rls-apse1-eks-a.risingwave-cloud.xyz:4566/dev?options=--tenant%3Drwc-3-sushant3node"`, and `flush` also get stuck. 🥵  could you please also attach the logs of all compute nodes? And check whether they're alive?

**Comment by HuaHuaY** [comment]
[risectl_compute_0_log.txt](https://github.com/risingwavelabs/risingwave/files/10122177/risectl_compute_0_log.txt)
[risectl_compute_1_log.txt](https://github.com/risingwavelabs/risingwave/files/10122178/risectl_compute_1_log.txt)
[risectl_compute_2_log.txt](https://github.com/risingwavelabs/risingwave/files/10122179/risectl_compute_2_log.txt)
Three computes' `risectl trace` logs.

**Comment by sushantgupta** [comment]
As per as discussion with @sumittal, this issue is fixed, we can close this.

**Comment by BugenZhao** [comment]
So what happened before? 🤔

**Comment by sumittal** [comment]
Ok, So there are 2 different issues:
1. this issue where I used the existing setup and somehow the commands were getting stuck but could not reproduce the same on other setups, and also did not encounter this issue again.
2. this issue is related to the materialized view creation state, where we cannot cancel it using Cntl-C which is already opened and scheduled for future to fix.

So, we can closed the first one unless can be reproduced. @BugenZhao
